### ISSUE TITLE
Stock - wannabe whole value?

### ISSUE THREAD

**Issue by DanielOlewski** [ericgu/AnExerciseInCouplingDesRefWG#12]
no validation on constructor inputs
dates as strings
negative double values accepted

**Comment by wli3** [comment]
Agree, stock.Ticker is being used a lot. If it is whole value, the field of it should not be accessed that much.
